controller/scheduler-v2/scheduler.go

```diff
@@ -0,0 +1,292 @@
+package main
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/flynn/flynn/Godeps/_workspace/src/gopkg.in/inconshreveable/log15.v2"
+	ct "github.com/flynn/flynn/controller/types"
+	"github.com/flynn/flynn/controller/utils"
+	"github.com/flynn/flynn/host/types"
+)
+
+type Scheduler struct {
+	utils.ControllerClient
+	utils.ClusterClient
+	log        log15.Logger
+	formations *Formations
+
+	jobs    *jobMap
```

Comment:
Yep totally understandable, I think starting broadly from the current scheduler and then cleaning things up is fine.

The reason the current scheduler has the states in two formats is because handling formation changes and job restarts is handled separately, we should be aiming to unify that somewhat.